authorization/permission/model/permission_model_service_blackbox_test.go

```diff
@@ -0,0 +1,687 @@
+		return nil, err
+	}
+
+	createdResource, err := s.resourceRepo.Load(s.Ctx, resource.ResourceID)
+	require.NoError(s.T(), err, "Could not load resource")
+	if err != nil {
+		return nil, err
+	}
+
+	return createdResource, nil
+}
+
+// Creates a child resource of the specified parent resource, of the specified type
+func (s *permissionModelServiceBlackBoxTest) createTestChildResource(parentResource resource.Resource, resourceTypeName string) (*resource.Resource, error) {
+	// Lookup the resource type
+	resourceType, err := s.resourceTypeRepo.Lookup(s.Ctx, resourceTypeName)
+	require.NoError(s.T(), err, "Could not lookup resource type")
+
+	resource := &resource.Resource{
+		ResourceID:       uuid.NewV4().String(),
+		ParentResourceID: &parentResource.ResourceID,
+		ResourceType:     *resourceType,
+		ResourceTypeID:   resourceType.ResourceTypeID,
+	}
+
+	err = s.resourceRepo.Create(s.Ctx, resource)
+	require.NoError(s.T(), err, "Could not create resource")
+	if err != nil {
+		return nil, err
+	}
+
+	createdResource, err := s.resourceRepo.Load(s.Ctx, resource.ResourceID)
+	require.NoError(s.T(), err, "Could not load resource")
+	if err != nil {
+		return nil, err
+	}
+
+	return createdResource, nil
+}
+
+// Creates a test organization identity
+func (s *permissionModelServiceBlackBoxTest) createTestOrganization(db *gorm.DB, appDB application.DB,
+	creatorIdentityID uuid.UUID, name string) (account.Identity, error) {
+
+	orgModelService := organizationModel.NewOrganizationModelService(db, appDB)
+	orgService := organizationService.NewOrganizationService(orgModelService, appDB)
+
+	var organization *account.Identity
+
+	orgID, err := orgService.CreateOrganization(s.Ctx, creatorIdentityID, name)
+	if err != nil {
+		return *organization, err
+	}
+
+	repo := account.NewIdentityRepository(db)
+
+	organization, err = repo.Load(s.Ctx, *orgID)
+	if err != nil {
+		return *organization, err
+	}
+
+	return *organization, nil
+}
+
+// Adds a member to the specified identity (i.e. organization)
+func (s *permissionModelServiceBlackBoxTest) addMember(db *gorm.DB, appDB application.DB,
+	memberOf uuid.UUID, memberId uuid.UUID) error {
+
+	// TODO replace this with the repository method once membership repo is implemented
+	db.Unscoped().Exec("INSERT INTO membership (member_of, member_id) VALUES (?, ?)", memberOf, memberId)
+
+	return nil
+}
+
+// Removes a member
+// TODO remove this when the membership repository is implemented
+func (s *permissionModelServiceBlackBoxTest) removeMember(db *gorm.DB, appDB application.DB,
+	memberOf uuid.UUID, memberId uuid.UUID) error {
+
+	db.Unscoped().Exec("DELETE FROM membership WHERE member_of = ? and member_id = ?", memberOf, memberId)
+	return nil
+}
+
+// Creates a role mapping
+func (s *permissionModelServiceBlackBoxTest) createRoleMapping(db *gorm.DB, appDB application.DB,
+	resourceID string, fromRoleID uuid.UUID, toRoleID uuid.UUID) error {
+	err := s.roleMappingRepo.Create(s.Ctx, &roleRepo.RoleMapping{
```

Comment:
might be useful for other tests too - so you might consider moving this to test/authorization.go

Reply:
Moved this also.